tezedge_state/src/proposer/mod.rs

```diff
@@ -0,0 +1,948 @@
+    //                 self.state,
+    //                 TickProposal {
+    //                     time_passed: self.update_time(at),
+    //                     effects: &mut self.effects,
+    //                 },
+    //                 self.proposal_persister
+    //             );
+    //         }
+    //         Event::Network(event) => {
+    //             self.handle_network_event(&event);
+    //         }
+    //     }
+    // }
+
+    fn handle_event_ref<'a>(&mut self, event: EventRef<'a, NetE>) {
+        match event {
+            Event::Tick(at) => {
+                accept_proposal!(
+                    self.state,
+                    TickProposal {
+                        time_passed: self.time.update(at).take_elapsed(),
+                        effects: &mut self.effects
+                    },
+                    self.proposal_persister
+                );
+            }
+            Event::Network(event) => {
+                self.handle_network_event(event);
+            }
+        }
+    }
+
+    fn handle_network_event(&mut self, event: &NetE) {
+        self.time.update(event.time());
+        if event.is_server_event() {
+            // we received event for the server (client opened tcp stream to us).
+            loop {
+                // as an optimization, execute requests only after 100
+                // accepted new connections. We need to execute those
+                // requests as they might include command to stop
+                // listening for new connections or disconnect new peer,
+                // if for example they are blacklisted.
+                for _ in 0..100 {
+                    match self.manager.accept_connection(&event) {
+                        Some(peer) => {
+                            accept_proposal!(
+                                self.state,
+                                NewPeerConnectProposal {
+                                    effects: &mut self.effects,
+                                    time_passed: self.time.take_elapsed(),
+                                    peer: peer.address().clone(),
+                                },
+                                self.proposal_persister
+                            );
+                            self.handle_readiness_event(event);
+                        }
+                        None => return,
+                    }
+                }
+                self.execute_requests();
+            }
+        } else {
+            self.handle_readiness_event(event);
+        };
+    }
+
+    fn handle_readiness_event(&mut self, event: &NetE) {
+        self.time.update(event.time());
+        let peer = match self.manager.get_peer_for_event_mut(&event) {
+            Some(peer) => peer,
+            None => {
+                // Should be impossible! If we receive an event for
+                // the peer, that peer must exist in manager.
+                // TODO: write error log.
```

Comment:
@binier 
please, better do that log (error/warn) now, we could see if that really happens and how often

Reply:
It's not really an issue even if we receive such an event, nothing will break, though ok I can add log in there.

Reply:
@binier 
yes, please, add here at least a log,
we should log any error from the begining and later can decide or fix if it occures too often